### ISSUE TITLE
laszip try to read too many points

### ISSUE THREAD

**Issue by Splifit** [laspy/laspy#212]
When trying to laspy.read() .laz files uploaded on : https://remotesensing.vlaanderen.be/apps/openlidar/

The following kind of error is redundant : 

laszip.LaszipError: reading point 1698081 of 1698070 total points

the read_n_points() routines always try to read too many points.  
The linked image is a resume of the point format used.

Nevertheless, This problem does not show when working with data from : https://portal.opentopography.org/datasets

I am a bit loss on what to do.

Regards,

Lucas

![Capture d’écran de 2022-04-16 14-36-40](https://user-images.githubusercontent.com/102418617/163675497-8fdba6da-ee40-4861-bccb-5ada018eafe1.png)

**Comment by tmontaigu** [comment]
I tried some of the files and did not have any errors, can you five a file where it happens ?

This error  likely means that the header of the las in incorrect  by saying that are more points that there actually are, or the file somehow got truncated during download ?

**Comment by Splifit** [comment]
You probably tried the files called DHMV I which does not generate the same error. The series of data which cause the problem is DHMV II : here is a link to download one such file : 
https://remotesensing.vlaanderen.be/download/openlidar/LiDAR_DHMV_2/P5/ATL12516/LiDAR_DHMV_2_P5_ATL12516_FU_20150327_16073_4/Tiles/LiDAR_DHMV_2_P5_ATL12516_FU_20150327_16073_4_171500_202500.laz 
(Github does not permit to send .laz through this discussion)

**Comment by tmontaigu** [comment]
The files have a problem, the number of points written in the header (so the number of points we expect and try to read) does not  match the reality and so we reach the end of file sooner than expected

**Comment by Splifit** [comment]
Ok thank you for your expertise. Do you have any roundabout to correct this ? 

I really need to be able to process the whole dataset Is there any routines that would permit me to change the header number with the correct one ?

**Comment by tmontaigu** [comment]
I don't think that would be possible with current laspy, I would need to provide ways to handle this kind of cases

The easiest way for now would be to use the command line `laszip` or `las2las` from `LAStools`, and for each file do
`las2las -i file_with_problem.laz -o corrected.laz` or `laszip -i file_with_problem.laz -o corrected.laz`

**Comment by Splifit** [comment]
Ok thank you very much for your help.

I will try to install LAStools and see if I can create a pipeline to solve this problem.

regards,

Lucas
________________________________
De : tmontaigu ***@***.***>
Envoyé : lundi 18 avril 2022 17:30
À : laspy/laspy ***@***.***>
Cc : Splifit ***@***.***>; Manual ***@***.***>
Objet : Re: [laspy/laspy] laszip try to read too many points (Issue #212)


I don't think that would be possible with current laspy, I would need to provide ways to handle this kind of cases

The easiest way for now would be to use the command line laszip or las2las from LAStools, and for each file do
las2las -i file_with_problem.laz -o corrected.laz or laszip -i file_with_problem.laz -o corrected.laz

**Comment by kl-eswar** [comment]
> You probably tried the files called DHMV I which does not generate the same error. The series of data which cause the problem is DHMV II : here is a link to download one such file : https://remotesensing.vlaanderen.be/download/openlidar/LiDAR_DHMV_2/P5/ATL12516/LiDAR_DHMV_2_P5_ATL12516_FU_20150327_16073_4/Tiles/LiDAR_DHMV_2_P5_ATL12516_FU_20150327_16073_4_171500_202500.laz (Github does not permit to send .laz through this discussion)

So, getting a LazBackend error when directly tried with laspy.
